pkg/flavor/kubernetes/nfs.go

```diff
@@ -453,44 +459,92 @@ func (flavor *Flavor) getNFSSpec(scParams map[string]string) (*NFSSpec, error) {
 	}
-	resourceLimits[core_v1.ResourceCPU] = cpuQuantity
+	resourceLimits[core_v1.ResourceCPU] = cpuLimitQuantity
 
-	// set factory default memory
-	memoryQuantity, err := resource.ParseQuantity(defaultRLimitMemory)
+	// set factory default memory limits
+	memoryLimitQuantity, err := resource.ParseQuantity(defaultRLimitMemory)
 	if err != nil {
-		return nil, fmt.Errorf("invalid nfs memory resource limit %s provided in defaults, err %s", defaultRLimitMemory, err.Error())
+		return nil, fmt.Errorf("Invalid NFS memory resource limit '%s' provided in defaults, err %s", defaultRLimitMemory, err.Error())
 	}
-	resourceLimits[core_v1.ResourceMemory] = memoryQuantity
+	resourceLimits[core_v1.ResourceMemory] = memoryLimitQuantity
 
-	// nfs cpu limits eg: cpu=500m
+	// User provided cpu limits eg: cpu=500m
 	if val, ok := scParams[nfsResourceLimitsCPUKey]; ok {
 		quantity, err := resource.ParseQuantity(val)
 		if err != nil {
-			return nil, fmt.Errorf("invalid nfs cpu resource limit %s provided in storage class, err %s", val, err.Error())
+			return nil, fmt.Errorf("Invalid '%s: %s' provided in StorageClass, err %s", nfsResourceLimitsCPUKey, val, err.Error())
 		}
 		resourceLimits[core_v1.ResourceCPU] = quantity
 	}
 
-	// nfs memory limits eg: memory=64Mi
+	// User provided memory limits eg: memory=64Mi
 	if val, ok := scParams[nfsResourceLimitsMemoryKey]; ok {
 		quantity, err := resource.ParseQuantity(val)
 		if err != nil {
-			return nil, fmt.Errorf("invalid nfs memory resource limit %s provided in storage class, err %s", val, err.Error())
+			return nil, fmt.Errorf("Invalid '%s: %s' provided in StorageClass, err %s", nfsResourceLimitsMemoryKey, val, err.Error())
 		}
 		resourceLimits[core_v1.ResourceMemory] = quantity
 	}
 
-	nfsSpec.resourceRequirements = &core_v1.ResourceRequirements{Limits: resourceLimits}
+	// Requests
+	resourceRequests := make(core_v1.ResourceList)
+
+	// set factory default cpu requests
+	cpuRequestQuantity, err := resource.ParseQuantity(defaultRRequestCPU)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid NFS CPU resource request %s provided in defaults, err %s", defaultRRequestCPU, err.Error())
+	}
+	resourceRequests[core_v1.ResourceCPU] = cpuRequestQuantity
+
+	// set factory default memory requests
+	memoryRequestQuantity, err := resource.ParseQuantity(defaultRRequestMemory)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid NFS memory resource request '%s' provided in defaults, err %s", defaultRRequestMemory, err.Error())
+	}
+	resourceRequests[core_v1.ResourceMemory] = memoryRequestQuantity
+
+	// User provided NFS cpu requests eg: cpu=500m
+	if val, ok := scParams[nfsResourceRequestsCPUKey]; ok {
+		quantity, err := resource.ParseQuantity(val)
+		if err != nil {
+			return nil, fmt.Errorf("Invalid '%s: %s' provided in StorageClass, err %s", nfsResourceRequestsCPUKey, val, err.Error())
+		}
+		resourceRequests[core_v1.ResourceCPU] = quantity
+	}
```

Comment:
Can we use the 'else' condition here to assign default value only when a user-provided value is absent, thereby avoiding redundant variable assignments?

Reply:
Thanks for calling out the sloppy code. I removed a lot of the duplicate code by replacing it with a function.